kms/kms.rb

```diff
@@ -0,0 +1,392 @@
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in write, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Note: Code samples in this file set constants, and
+#       constants cannot be set inside method definitions in Ruby,
+#       code snippets in this sample are wrapped in global lambdas.
+
+$create_keyring = -> (project_id:, key_ring_id:, location:) do
+  # [START kms_create_keyring]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}"
+
+  # Create a KeyRing for your project
+  key_ring = kms_client.create_project_location_key_ring(resource,
+      Cloudkms::KeyRing.new, key_ring_id: key_ring_id)
+
+  puts "Created KeyRing #{key_ring_id}"
+  # [END kms_create_keyring]
+end
+
+$create_cryptokey = -> (project_id:, key_ring_id:, crypto_key:, location:) do
+  # [START kms_create_cryptokey]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # crypto_key  = "Name of cryptoKey"
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}"
+
+  # Create a CryptoKey for your project keyring
+  new_crypto_key = kms_client.create_project_location_key_ring_crypto_key(resource,
+      Cloudkms::CryptoKey.new(purpose: "ENCRYPT_DECRYPT"), crypto_key_id: crypto_key)
```

Comment:
As mentioned above, I'd love to see these 2 params on their own separate lines.

(*even tho it will take up more vertical space... I find this a little hard to read, as is*)